cmake/get_info.cmake

```diff
@@ -0,0 +1,310 @@
+SET(detailed_log "${CMAKE_BINARY_DIR}/detailed.log")
+SET(minimal_log  "${CMAKE_BINARY_DIR}/minimal.log")
+FILE(REMOVE ${detailed_log} ${minimal_log})
+
+MACRO(_both)
+  # Write to both log files:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_detailed)
+  # Only write to detailed.log:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_minimal)
+  # Only write to minimal.log:
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+FUNCTION(build_type_spec log_type var_name)
+    if(CMAKE_BUILD_TYPE MATCHES "Release")
+        set(var_string "${var_name}_RELEASE")
+        set(upd_string "\n#        ${var_name}_RELEASE                   ${${var_string}} ")
+        FILE(APPEND ${log_type} ${upd_string})
+    endif()
+    if(CMAKE_BUILD_TYPE MATCHES "Debug" )
+        set(var_string "${var_name}_DEBUG")
+        set(upd_string "#       ${var_name}_DEBUG                 ${${var_string}} ")
+        FILE(APPEND ${log_type} "${upd_string}")
+    endif()
+    if(CMAKE_BUILD_TYPE MATCHES "RelWithDebInfo")
+        set(var_string "${var_name}_RELWITHDEBINFO")
+        set(upd_string "#       ${var_name}_RELWITHDEBINFO                      ${${var_string}} ")
+        FILE(APPEND ${log_type} "${upd_string}")
+    endif()
+    if(CMAKE_BUILD_TYPE MATCHES "MinSizeRel")
+        set(var_string "${var_name}_MINSIZEREL")
+        set(upd_string "#       ${var_name}_MINSIZEREL                  ${${var_string}} ")
+        FILE(APPEND ${log_type} "${upd_string}")
+    endif()
+ENDFUNCTION()
+
+_both(
+"
+
+
+"
+)
+
+IF("${GINKGO_GIT_SHORTREV}" STREQUAL "")
+    _both("
+#########################################################################
+#
+#  Summary of Configuration for  (Ginkgo version ${Ginkgo_VERSION_TAG})\n")
+ELSE()
+    _both("
+#########################################################################
+#
+#  Summary of Configuration for  (Ginkgo version ${Ginkgo_VERSION_TAG}, shortrev ${GINKGO_GIT_SHORTREV})\n")
+ENDIF()
+_both(
+"#
+#
+#  Ginkgo configuration:
+#        CMAKE_BUILD_TYPE:                       ${CMAKE_BUILD_TYPE}
```

Comment:
suggestion: We might be able to simplify this into `ginkgo_print_variable(CMAKE_BUILD_TYPE)` to avoid potential typos.

Reply:
Yes, I was a bit lazy. :P Should be done now.